runtime/client_http_request.go

```diff
@@ -56,13 +56,20 @@ func NewClientHTTPRequest(
 	clientID, methodName string,
 	client *HTTPClient,
 ) *ClientHTTPRequest {
+	scope := client.Scope.Tagged(map[string]string{
+		"client": clientID,
+		"method": methodName,
+	})
+
+	scopeFields := GetScopeFieldsFromCtx(ctx)
+	scope = scope.Tagged(scopeFields)
 	req := &ClientHTTPRequest{
 		ClientID:       clientID,
 		MethodName:     methodName,
 		client:         client,
 		Logger:         client.loggers[methodName],
 		ContextLogger:  NewContextLogger(client.loggers[methodName]),
-		metrics:        client.metrics[methodName],
+		metrics:        NewOutboundHTTPMetrics(scope),
```

Comment:
This probably will make a performance hit because now it creates sub scopes and metric objects per request while it is previously per client method. It does make sense because we are extracting scope tags dynamically at runtime. And I can't think of an easy way to get around it as the tags are unknown at compile time. For now let's put a todo comment here to document the impact on performance so that we know where to revisit in the future.